nautobot_golden_config/utilities/helper.py

```diff
@@ -118,58 +117,12 @@ def render_jinja_template(obj, logger, template):
         raise NornirNautobotException from error
 
 
-def clean_config_settings(repo_type: str, repo_count: int, match_rule: str):
-    """Custom clean for `GoldenConfigSettingFeatureForm`.
-
-    Args:
-        repo_type (str): `intended` or `backup`.
-        repo_count (int): Total number of repos.
-        match_rule (str): Template str provided by user to match repos.
-
-    Raises:
-        ValidationError: Custom Validation on form.
-    """
-    if repo_count > 1:
-        if not match_rule:
-            raise forms.ValidationError(
-                f"If you specify more than one {repo_type} repository, you must provide the {repo_type} repository matching rule template."
-            )
-    elif repo_count == 1 and match_rule:
-        raise forms.ValidationError(
-            f"If you configure only one {repo_type} repository, do not enter the {repo_type} repository matching rule template."
-        )
+def get_device_to_settings_map(queryset):
```

Comment:
This seems pretty inefficient

Reply:
To some extent, Yes. In Python, dictionaries contain pointers (not copies of actual objects). What else would be a concern?

Reply:
in order to speed up the execution of this function we could just query the IDs of the devices and not the entire object. 
something like that

```python
    queryset_ids = queryset.values_list('id', flat=True)
    for golden_config_setting in models.GoldenConfigSetting.objects.all():
        for device_id in golden_config_setting.get_queryset().values_list('id', flat=True):
            if (device_id in queryset_ids) and (device_id not in device_to_settings_map):
                device_to_settings_map[device_id] = golden_config_setting
```